src/routes/docs/products/functions/develop-locally/+page.markdoc

```diff
@@ -0,0 +1,114 @@
+
+{% arrow_link href="/docs/tooling/command-line/functions" %}
+Learn more about deploying functions using CLI
+{% /arrow_link %}
+
+# Locally develop your function {% #locally-develop-your-function %}
+
+Use the `appwrite run functions` command to develop your function locally.
+
+```sh
+appwrite run functions
+? Which function would you like to develop locally? My Awesome Function ("<FUNCTION_ID>")
+
+runtime   │ entrypoint  │ path                           │ commands     
+───────────┼──────────────┼───────────────────────────────────┼─────────────
+node-16.0 │ src/main.js │ functions/"<FUNCTION_ID>"        │ npm install
+
+ℹ Info: If you wish to change your local settings, update the appwrite.json file and rerun the 'appwrite run' command.
+♥ Hint: Permissions, events, CRON and timeouts dont apply when running locally.
+ℹ Info: Pulling Docker image ...
+♥ Hint: This may take a few minutes, but we only need to do this once.
+ℹ Info: Building function using Docker ...
+Preparing for build ...
+
+Building ...
+
+
+added 4 packages, and audited 5 packages in 2s
+
+
+1 package is looking for funding
+  run `npm fund` for details
+
+
+found 0 vulnerabilities
+
+Packing build ...
+
+Build finished.
+
+ℹ Info: Starting function using Docker ...
+♥ Hint: Function automatically restarts when you edit your code.
+✓ Success: Visit http://localhost:3000/ to execute your function.
+```
+
+This command helps you efficiently develop your Appwrite functions on your local machine. When developing your Appwrite function locally, it will receive headers like when you develop a function in the Console.
+
+{% arrow_link href="/docs/products/functions/develop" %}
+Learn more about developing a function
+{% /arrow_link %}
+
+# Headers {% #headers %}
+
+Appwrite functions receive various headers, like `x-appwrite-key` and `x-appwrite-event`. To view all the received headers, use the following:
+
+```javascript
+JSON.stringify(req.headers); // String key-value pairs of all request headers, keys are lowercase
+```
+
+An example of the function headers response:
+
+```sh
+{
+  "host": "localhost:3000",
+  "connection": "keep-alive",
+  "sec-ch-ua": "\"Google Chrome\";v=\"127\"",
+  "sec-ch-ua-mobile": "?0",
+  "user-agent": "Chrome/127.0.0.0 Safari/537.36",
+  "sec-ch-ua-platform": "\"Windows\"",
+  "accept": "image/avif,image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8",
+  "sec-fetch-site": "same-origin",
+  "sec-fetch-mode": "no-cors",
+  "sec-fetch-dest": "image",
+  "referer": "http://localhost:3000/",
+  "accept-encoding": "gzip, deflate, br, zstd",
+  "accept-language": "en-US,en;q=0.9",
+  "cookie": "<COOKIE>",
+  "x-appwrite-trigger": "http",
+  "x-appwrite-event": "",
+  "x-appwrite-user-id": "",
+  "x-appwrite-user-jwt": "<JWT_TOKEN>"
+}
+```
```

Comment:
I didn't quite mean to include a section like this. When writing functions, it's common to need to use a dynamic API key or impersonate a user using the JWT. We need to clearly address how things like that work when developing locally.